packages/front-end/components/PowerCalculation/stats.ts

```diff
@@ -0,0 +1,350 @@
+      Math.log(Math.sqrt(n * Math.pow(rho, 2) + 1) / alpha)) /
+    Math.pow(n * rho, 2);
+  const zSequential = Math.sqrt(n) * Math.sqrt(partUnderRadical);
+  const zStar = normal.quantile(1.0 - 0.5 * alpha, 0, 1);
+  const standardErrorSequential =
+    (standardErrorSampleMean * zSequential) / zStar;
+  const vSequential = Math.pow(standardErrorSequential, 2);
+  return n * vSequential;
+}
+
+export function sequentialPowerStandardError(
+  variance: number,
+  mean: number,
+  n: number,
+  nVariations: number,
+  effectSize: number,
+  alpha: number,
+  sequentialTuningParameter: number,
+): number {
+  const v = sequentialPowerSequentialVariance(
+    variance,
+    n / nVariations,
+    alpha,
+    sequentialTuningParameter,
+  );
+  return Math.sqrt(
+    frequentistVariance(
+      v,
+      mean,
+      n / nVariations,
+      v,
+      mean * (1.0 + effectSize),
+      n / nVariations,
+      true,
+    ),
+  );
+}
+
+/**
+ * Performs power calculation
+ *
+ * @param effectSize Scalar lift (relative to the scalar mean of the distribution, expressed as percentage).
+ * @param mean Scalar mean of the distribution.
+ * @param variance Scalar variance of the distribution.
+ * @param n Scalar sample size.
+ * @param nVariations Scalar number of variations.
+ * @param alpha false positive rate (default: 0.05).
+ * @param twoTailed Binary indicator if the test is 1 or 2-tailed (default: true).
+ * @returns Estimated power.
+ */
+export function powerEst(
+  effectSize: number,
+  mean: number,
+  variance: number,
+  n: number,
+  nVariations: number,
+  alpha: number = 0.05,
+  twoTailed: boolean = true,
+  sequentialTuningParameter = 0,
+): number {
+  if (typeof twoTailed !== "boolean") {
+    throw new Error("twoTailed must be boolean.");
+  }
+  const zStar = twoTailed
+    ? normal.quantile(1.0 - 0.5 * alpha, 0, 1)
+    : normal.quantile(1.0 - alpha, 0, 1);
+
+  let standardError = 0;
+  if (sequentialTuningParameter > 0) {
+    standardError = sequentialPowerStandardError(
+      variance,
+      mean,
+      n,
+      nVariations,
+      effectSize,
+      alpha,
+      sequentialTuningParameter,
+    );
+  } else {
+    standardError = powerStandardError(
+      variance,
+      mean,
+      n,
+      nVariations,
+      effectSize,
+    );
+  }
+  const standardizedEffectSize = effectSize / standardError;
+  const upperCutpoint = zStar - standardizedEffectSize;
+  let power = 1 - normal.cdf(upperCutpoint, 0, 1);
+  if (twoTailed) {
+    const lowerCutpoint = -zStar - standardizedEffectSize;
+    power += normal.cdf(lowerCutpoint, 0, 1);
+  }
+  return power;
+}
+
+/**
+ * Calculates minimum detectable effect
+ *
+ * @param power desired power.
+ * @param mean Scalar mean of the distribution.
+ * @param variance Scalar variance of the distribution.
+ * @param n Scalar sample size.
+ * @param nVariations Scalar number of variations.
+ * @param alpha false positive rate (default: 0.05).
+ * @returns Estimated power.
```

Comment:
nit: this comment describing the method is a bit out of date

Reply:
udpated comment